src/Components/test/testassets/Components.WasmMinimal/Program.cs

```diff
@@ -0,0 +1,10 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Reflection;
+using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
+
+Assembly.Load(nameof(TestContentPackage));
```

Comment:
Without this, the `TestContentPackage` assembly does not get loaded, and [`RootComponentTypeCache`](https://github.com/dotnet/aspnetcore/blob/6575b289081b9f85fb30d01c7b306d1ec5d00a68/src/Components/Shared/src/RootComponentTypeCache.cs#LL24C22-L24C22) can't discover the component.

Would we consider updating `RootComponentTypeCache` to attempt to load an assembly if it couldn't be found?

Reply:
This is very interesting. Is this just a quirk of the framework that is already the case in .NET 7? For example, if you used `<component ... render-mode="WebAssembly">` (as opposed to declaring the root component in `Program.cs`) could you get the same issue? My guess is yes.

Rather than block your PR on it, I would recommend filing an issue for us to dig into (unless you think you are actively *causing* this issue in your PR here, which doesn't appear to be the case to me).

> Would we consider updating RootComponentTypeCache to attempt to load an assembly if it couldn't be found?

Assuming you mean code that is actually running on WebAssembly (e.g., inside `WebAssemblyHostBuilder`) then yes sure, why not. It's not a security concern in that case.

There could still be a problem since trimming may have removed the component if there was no statically-observable evidence that it was going to be used. However in practice that would be uncommon since we don't trim user assemblies by default. This may change in the future.